src/components/StudyRoom/StudyRoom.js

```diff
@@ -0,0 +1,105 @@
+                    <Typography variant="body2">Created on: {roomData.createdAt ? roomData.createdAt.split("T")[0] : ''}</Typography>
+                </div>
+            </Stack>
+        </div>
+    )
+
+    const chatRoom = React.useMemo(() => (
+        <React.Fragment>
+            <NavDrawer/>
+
+            {/* Top */}
+            <StudyRoomChatCard width='92vw' height='10vh' marginTop='70px' topLeftRadius='10px' topRightRadius='10px'
+                               bottomLeftRadius='0px' bottomRightRadius='0px' content={roomHeader}/>
+
+            { /*  Chat  */ }
+            <StudyRoomChatCard width='92vw' height='65vh' marginTop='2px' topLeftRadius='0px' topRightRadius='0px'
+                               bottomLeftRadius='0px' bottomRightRadius='0px' content={<GetStudyRoomChat />}/>
+
+            { /* Typing area */ }
+            <StudyRoomChatCard width='92vw' height='5vh' marginTop='2px' topLeftRadius='0px' topRightRadius='0px'
+                               bottomLeftRadius='0px' bottomRightRadius='0px' content={<ChatFooter />}/>
+            <div style={{display: 'flex', flexDirection: 'row', marginLeft: '1.8vw', marginRight: '1.8vw'}}>
+
+                { /* Drawer left icon */ }
+                <StudyRoomChatCard width='30.5vw' height='7vh' marginTop='2px' topLeftRadius='0px' topRightRadius='0px'
+                                   bottomLeftRadius='10px' bottomRightRadius='0px' content={<div
+                    style={{width: '100%', height: '100%', background: 'none', border: 'none'}}
+                ><BottomDrawer icon={<DescriptionIcon style={{color: '#912338', height: '4vh', width: '4vh'}}/>}
+                               title={'Course Notes'} content={<CourseNotes />}/></div>}/>
+
+                { /* Drawer middle icon */ }
+                <StudyRoomChatCard width='30.5vw' height='7vh' marginTop='2px' topLeftRadius='0px' topRightRadius='0px'
+                                   bottomLeftRadius='0px' bottomRightRadius='0px' content={<div
+                    style={{width: '100%', height: '100%', background: 'none', border: 'none'}}
+                ><BottomDrawer icon={<GroupsIcon style={{color: '#912338', height: '6vh', width: '6vh'}}/>}
+                               title={'Participants'} content={<ParticipantsList/>}/></div>}/>
+
+                { /* Drawer right icon */ }
+                <StudyRoomChatCard width='30.5vw' height='7vh' marginTop='2px' topLeftRadius='0px' topRightRadius='0px'
+                                   bottomLeftRadius='0px' bottomRightRadius='10px' content={<div
+                    style={{width: '100%', height: '100%', background: 'none', border: 'none'}}
+                ><BottomDrawer icon={<SettingsIcon style={{color: '#912338', height: '4vh', width: '4vh'}}/>}
+                               title={'settings page'} content={<StudyRoomSettings/>}/>  asda  </div>}/>
+            </div>
+        </React.Fragment>
+    ), [roomData]);
```

Comment:
I don't think we want to re render this whole const when room data change. maybe we should split these into smaller const (one for each section). and re render these sections only when specific data (i.e ChatRoom id, messages, etc)  changes